### PR TITLE
Add escape and unescape functions

### PR DESCRIPTION
**Pull request by theScottyJam**
This PR was derived from https://github.com/theScottyJam/snap.js/pull/6

### CODE REVIEW

**Comment by crhallberg** [comment]
This is a much more thorough version of what I started with my PR! I was just looking for a replacement for an escaping function in another project, so I was happy with Lodash's version. You've taken it way farther!

If it would be useful, [I do have an escape function for HTML attributes](https://github.com/vufind-org/vufind/blob/dev/themes/bootstrap3/js/common.js#L390) from the project, if you would like to include this as a safe option for URLs and attributes. _EDIT_: Looking into your sources more, it looks like [`.setAttribute` works well for many attributes](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#safe-sinks), ~~but not URL ones~~ so that would likely replace this whole function. I'm learning in real time! Thank you!

File: content/nolodash/unescape/description.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    .replaceAll('&gt;', '>')
+    .replaceAll('&quot;', '"')
+    .replaceAll('&#39;', "'")
+    .replaceAll('&amp;', '&');
+}
+```
+
+## Don't use unescapeHtmlChars() unless you have to
+
+The use of `unescapeHtmlChars()` may (or may not) indicate an issue with how you're storing and transfering data. Escaping HTML characters is an operation that is generally done the moment before you insert it into your page. If this pattern is followed, you will generally have an unescaped version of your string on-hand as well, which removes the need for using an `unescapeHtmlChars()` function. You can read more about when to escape your HTML characters [here](https://security.stackexchange.com/questions/32394/when-to-escape-user-input/32396#32396).
```
Comment (crhallberg) [inline content/nolodash/unescape/description.md]: For accessibility, it's better to include more context in your link text. Many screen-reader users will list links available on a page and something like "read more about when to escape your HTML characters" is much better than "here".

Comment (crhallberg) [reply content/nolodash/unescape/description.md]: The rest of the links look great so maybe I protest too much!

File: README.md [hunk 2]
```diff
@@ -6,19 +6,40 @@ View the built website [over here](https://thescottyjam.github.io/snap.js/#!/nol
...
+* If two solutions have the same big-O, show the one that's more maintainable. In other words, don't micro-optimize. If two solutions have different big-Os, still favor showing the one that's more maintainable, but maybe leave a note about how there's room to improve its performance, or you could even show both solutions, letting the user pick the one that suits them best.
+* Lodash's implementations don't always follow modern best practices. Some examples:
+  * They like to overload their functions to have many different behaviors depending on the types of data you pass in (to a degree, this is fine, but sometimes its taken too far). It may be best to think of these as different problems being solved in the same function, which means you'd need to provide a list of solution for each problem being solved.
+  * They have a habbit of making parameters optional that really shouldn't be optional. If you can't think of a sensible reason for someone to purposefully omit a parameter, then don't bother supporting that "use case" in your non-lodash replacement.
```
Comment (crhallberg) [inline README.md]: Spellcheck for "habit".

**Comment by theScottyJam** [comment]
Thanks for the feedback!

**Pull request merged**